types/ember/index.d.ts

```diff
@@ -924,6 +935,11 @@ namespace Ember {
         static create(): MapWithDefault;
     }
     class Mixin<T, Base = Ember.Object> {
+        /**
+         * Mixin needs to have *something* on its prototype, otherwise it's treated like an empty interface.
+         */
+        __ember_mixin__: never;
```

Comment:
FWIW the TS compiler itself uses the term `brand` for this pattern, which they have sprinkled around pretty liberally e.g. https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts#L621